core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -316,6 +333,11 @@ public synchronized IndexService createIndex(String sIndexName, @IndexSettings S
         modules.add(new IndexAliasesServiceModule());
         modules.add(new IndexModule(indexSettings));
 
+        for (Module pluginModule : pluginsService.indexModules(indexSettings)) {
```

Comment:
is this actually used anywhere? I mean why would you add your own index modules? I really wonder if we need this kind of extension mechanism? I know it's unrelated but maybe we need a followup issue?